### PR TITLE
docs(ci): expand introduction to CI workflows

### PR DESCRIPTION
**Pull request by gancerlory**
Update the README with information from this folder: https://github.com/paradigmxyz/reth/tree/main/.github/workflows. If there are any inaccuracies in the descriptions of the YAML files, feel free to correct them.

### CODE REVIEW

File: docs/repo/ci.md [hunk 1]
```diff
@@ -16,10 +19,37 @@ The CI runs a couple of workflows:
...
 - **[release]**: Runs the release workflow
+- **[release-dist]**: Publishes Reth to external package managers
+- **[dependencies]**: Runs `cargo update` periodically to keep dependencies current
+- **[stale]**: Marks issues as stale if there has been no activity
+
+### Maintenance
+
+- **[docker]**: Publishes the Docker image.
+- **[assertoor]**: Runs Assertoor tests on Reth pairs.
+- **[hive]**: Runs `ethereum/hive` tests.
```
Comment (Rjected) [inline docs/repo/ci.md:30]: I would split it like this
```suggestion
- **[release-dist]**: Publishes Reth to external package managers
- **[dependencies]**: Runs `cargo update` periodically to keep dependencies current
- **[stale]**: Marks issues as stale if there has been no activity
- **[docker]**: Publishes the Docker image.

### Integration Testing

- **[assertoor]**: Runs Assertoor tests on Reth pairs.
- **[hive]**: Runs `ethereum/hive` tests.
```

**Review by Rjected — CHANGES_REQUESTED** [review]
thanks! I have one comment otherwise this is great

File: docs/repo/ci.md (hunk 1 above)
Comment (gancerlory) [reply docs/repo/ci.md]: Nice!

**Review by Rjected — APPROVED** [review]
lgtm, thanks!

**Pull request merged**